openverse_catalog/dags/providers/provider_workflow_types.py

```diff
@@ -0,0 +1,167 @@
+                       current execution date the `main_function` should
+                       be run (if `dated=True`).
+    execution_timeout: datetime.timedelta giving the amount of time a given data
+                       pull may take.
+    doc_md:            string which should be used for the DAG's documentation markdown
+    media_types:       list describing the media type(s) that this provider handles
+                       (e.g. `["audio"]`, `["image", "audio"]`, etc.)
+    """
+
+    dag_id: str
+    provider_script: str
+    default_args: Optional[Dict] = None
+    start_date: datetime = datetime(1970, 1, 1)
+    max_active_tasks: int = 1
+    schedule_string: str = "@monthly"
+    dated: bool = False
+    day_shift: int = 0
+    execution_timeout: timedelta = timedelta(hours=24)
+    doc_md: str = ""
+    media_types: Sequence[str] = ("image",)
+
+
+PROVIDER_WORKFLOWS = [
+    ProviderWorkflow(
+        dag_id="brooklyn_museum_workflow",
+        provider_script="brooklyn_museum",
+        start_date=datetime(2020, 1, 1),
+    ),
+    ProviderWorkflow(
+        dag_id="cleveland_museum_workflow",
+        provider_script="cleveland_museum_of_art",
+        start_date=datetime(2020, 1, 15),
+        execution_timeout=timedelta(hours=12),
+    ),
+    ProviderWorkflow(
+        dag_id="europeana_workflow",
+        provider_script="europeana",
+        schedule_string="@daily",
+        dated=True,
+    ),
+    ProviderWorkflow(
+        dag_id="finnish_museums_workflow",
+        provider_script="finnish_museums",
+        start_date=datetime(2020, 9, 1),
+        execution_timeout=timedelta(days=3),
+    ),
+    ProviderWorkflow(
+        dag_id="flickr_wokrflow",
+        provider_script="flickr",
+        schedule_string="@daily",
+        dated=True,
+    ),
+    ProviderWorkflow(
+        dag_id="freesound_workflow",
+        provider_script="freesound",
+        # doc_md = freesound.__doc__,
+        media_types=("audio",),
+    ),
+    ProviderWorkflow(
+        dag_id="jamendo_workflow",
+        provider_script="jamendo",
+        media_types=("audio",),
+    ),
+    ProviderWorkflow(
+        dag_id="metropolitan_museum_workflow",
+        provider_script="metropolitan_museum_of_art",
+        schedule_string="@daily",
+        dated=True,
+        execution_timeout=timedelta(hours=12),
+    ),
+    ProviderWorkflow(
+        dag_id="museum_victoria_workflow",
+        provider_script="museum_victoria",
+        start_date=datetime(2020, 1, 1),
+    ),
+    ProviderWorkflow(
+        dag_id="nypl_workflow",
+        provider_script="nypl",
+        start_date=datetime(2020, 1, 1),
+    ),
+    ProviderWorkflow(
+        dag_id="phylopic_workflow",
+        provider_script="phylopic",
+        schedule_string="@weekly",
+        dated=True,
+        execution_timeout=timedelta(hours=12),
+    ),
+    ProviderWorkflow(
+        dag_id="rawpixel_workflow",
+        provider_script="raw_pixel",
```

Comment:
Hmm, perhaps we should rename the rawpixel script to be `rawpixel` instead for consistency 😅